Lesson 6

Botanik1592 opened the pull request.

Botanik1592 committed:
Ajax page reload

Botanik1592 committed:
AJAX testing

vkurennov commented on app/controllers/answers_controller.rb:
```diff
@@ -8,7 +8,6 @@ def create
     if @answer.save
-      redirect_to @question
     else
       render 'questions/show'
```
Нет смысла делать рендер в ajax-запросах, т.к. юзер все равно ничего не увидит, ответ же в фоне придет. Поэтому обработку ошибок надо делать в js-ответе, но об этом в след. уроке, пока оставь так.

vkurennov commented on app/views/answers/create.js.erb:
```diff
@@ -0,0 +1,2 @@
+$('.answers').html('<%= j render 'questions/answers' %>');
```
1. По заданию, нужно не весь список рендерить, а добавлять только созданный ответ
2. Вспоминай Базовый курс и тему про анонимные паршалы. Сделай паршл answers/_answer  для 1 ответа, тогда здесь отрендерить ответ можно так: `render @answer`, а там, где нужен список, так: `render @question.answers`

vkurennov reviewed (changes requested):
Исправь замечания

Botanik1592 committed:
Add answers/_answer.html.slim
Remove questions/_answers.html.slim
Fix create.js.erb
Fix AnswerController test

Botanik1592 committed:
Fix render

Botanik1592 committed:
Add 'render answer create template' to answers_controller_spec
